gemd/units/impl.py

```diff
@@ -75,9 +137,9 @@ def _format_clean(unit, registry, **options):
     numerator = []
     denominator = []
     for u, p in unit.items():
-        if re.match(r"_[\d_]+$", u):
-            # Munged scaling factor; drop leading underscore, restore . and -
-            u = re.sub(r"(?<=\d)_(?=\d)", ".", u[1:]).replace("_", "-")
+        if re.match(r"_[\d_]+", u):
+            # Munged scaling factor; grab symbol, which is the prettier
+            u = registry.get_symbol(u)
```

Comment:
Use the symbol rather than trying to un-munge a number with regular expressions.